Add livereload invoke task. Fixes #1326

johnfraney opened the pull request:
Adds a `livereload` invoke task that reloads the browser window when
content files are updated.

Usage:

```console
$ invoke livereload
[I 190202 16:28:30 server:298] Serving on http://127.0.0.1:5500
[I 190202 16:28:30 handlers:59] Start watching changes
[I 190202 16:28:30 handlers:61] Start detecting changes
[I 190202 16:28:32 handlers:132] Browser Connected: http://127.0.0.1:5500/
```

See: https://livereload.readthedocs.io/en/latest/

johnfraney commented:
I'll rebase this once #2528 is pulled in

bhrutledge commented:
This looks handy, but one possible gotcha is that python-livereload [doesn't detect file deletion](https://github.com/lepture/python-livereload/issues/197) on macOS/Windows, which I discovered while implementing a [workaround](https://merlijn.vandeen.nl/2015/pelican-livereload.html) suggested in #1326.

justinmayer commented:
@bhrutledge: Thank you for [addressing the file deletion detection problem in python-livereload](https://github.com/lepture/python-livereload/pull/198) you mentioned above. 🎊 

@johnfraney: Many thanks for your contribution. Would you please rebase this PR on current master?

johnfraney commented:
@justinmayer Happy to help! I just rebased and added a call to the `build` task within the `livereload` task so it builds the project right away instead of only waiting for a file to change.

johnfraney commented:
@justinmayer I always get sad when I see those failing CI checks. I opened #2570 to fix some CI issues on master.

justinmayer commented:
@johnfraney: I re-launched the build for this PR, and Travis is reporting test failures due to `live-reload` needing Tornado, which needs SSL support in Python 2.7.9+... or something to that effect.

In any case, I don't think we need to add `live-reload` to `requires` in [setup.py](https://github.com/getpelican/pelican/pull/2526/files#diff-2eeaed663bd0d25b7e608891384b7298R11). The reason is that we try to keep hard dependencies to a minimum. For example, we don't even require Markdown to be installed, since folks who only use reStructuredText don't need it. I think it's best to instead inform users that they'll need to manually `pip install live-reload` in order to use this Invoke task.

As far as where to put those instructions... You had asked before about documentation, and I apologize for not addressing that question until now. Given that we are talking about automation via Invoke, I think the best place would be that [section of the documentation](https://docs.getpelican.com/en/latest/publish.html#invoke).

johnfraney commented:
@justinmayer I just sent up another commit that should implement your feedback. I removed the hard `livereload` dependency and added a paragraph about the new task to the Invoke section of the docs. Let me know if this needs anything else, and I'll get it added in.

justinmayer commented:
Well done, John. Glad to have this added to the project. Many thanks for your efforts and contribution! 🎊

johnfraney commented:
@justinmayer You're welcome! I've enjoyed using Pelican, and I'm happy I can give back a little.

talha131 commented:
@johnfraney Thanks for adding this feature to Pelican. It's very handy and a time saver.

johnfraney commented:
@talha131 I'm glad you think so! It's been treating me well for the last few months. It's a nice quality-of-life improvement.

ashb commented:
Slightly off topic to this issue (but I'm trying to use it) but I'm upgrading from an old 3.x install and don't have a tasks.py so I cant run `invoke` successfully. Is there a recommend way of generating that file?

justinmayer commented:
Run `pelican-quickstart` in an empty directory, answering the questions as usual. Then copy the relevant bits from the generated tasks.py file over to the tasks.py file in your site project.
